state/state_test.go

```diff
@@ -5225,3 +5225,29 @@ func (s *StateSuite) TestPeerRelationCreatesApplicationSettings(c *gc.C) {
 	_, err = settings.ReadSettings(key)
 	c.Assert(err, jc.ErrorIsNil)
 }
+
+func (s *StateSuite) TestAddRelationAssertsRelationCount(c *gc.C) {
```

Comment:
I'd expect a test to check this error was hit:
`errors.QuotaLimitExceededf("establishing a new relation for %s:%s would exceed its maximum relation limit of %d", ep.ApplicationName, ep.Relation.Name, ep.Relation.Limit)`

Reply:
I agree, but I think it should be a separate test. This one is checking the assertions.